Changed num <--> bytes conversion according to new Rust syntax

ameya-deshmukh opened the pull request:
Changed the num to bytes conversion (Little Endian) according to the latest Rust syntax. However, deserialization is left. @Stentonian wanted your thoughts on this as the deserialization involves other libraries as well, where I'll have to implement ```from_le_bytes``` separately. A bit more effort than the current PR, but doable. Should I have a go at it?

Stentonian commented:
Can you write a test to check the serialized outputs and then run the test against both the old and new code to see that they output the same values

ameya-deshmukh commented:
@Stentonian done. I actually realized that the serialization tests are already written in ```test_serialization``` and that they effectively check whether the serialization and deserialization works properly. Earlier, it was failing with my implementation, but I realised there was a subtle bug. Fixed it and now the test works fine. I have renamed it as ```test_new_serialization``` with the exact same functionality as before but to highlight the fact that the new implementation is correct as well. Please have a look, cheers.

Stentonian commented on Cargo.toml:
```diff
@@ -23,6 +23,7 @@ merlin = "3.0.0"
 
 thiserror = "1.0"
 displaydoc = "0.2"
+byteorder = "1.4.3"
```
What do you need this for? I seem to be able to use the `to_le_bytes()` without this import

Stentonian commented on src/range/padding.rs:
```diff
@@ -40,12 +39,14 @@ impl Serializable for RangeProofPadding {
     fn serialize(&self) -> Vec<u8> {
         let mut result: Vec<u8> = Vec::new();
         let mut bytes = self.get_aggregated().to_bytes();
-        result.append(&mut usize_to_bytes(bytes.len(), PROOF_SIZE_BYTE_NUM));
```
Removing `PROOF_SIZE_BYTE_NUM` check here is fine because it checks that the result is < 2^64 bits which is the limit of the usize type anyway, so the input could never be greater than that...

Stentonian commented on src/range/splitting.rs:
```diff
@@ -38,20 +36,23 @@ impl Serializable for RangeProofSplitting {
-            self.get_aggregated().len(),
-            AGGREGATED_NUM_BYTE_NUM,
-        ));
+        let len=self.get_aggregated().len() as u16;
```
...but changing `AGGREGATED_NUM_BYTE_NUM` check to u16 cast is not the same. Rust will allow an overflow i.e.
```rust
let a: usize = 2_usize.pow(2*8); // 65536
let b = a as u16; // 0
```
For now let's keep the panic instead of allowing overflow, and we can adjust the code later to remove the panic by changing the input type from usize to something else.

Stentonian commented:
Closing as this is stale
